### PR TITLE
replace ListStatuses API call with GetCombinedStatus when possible

### PR DESCRIPTION
**Pull request by adshmh**
Addresses #12265

### CODE REVIEW

File: prow/plugins/skip/skip_test.go [hunk 1]
```diff
@@ -183,7 +183,8 @@ func TestSkipStatus(t *testing.T) {
 					},
 				},
 			},
-			sha: "shalala",
+			sha:            "shalala",
+			combinedStatus: github.StatusSuccess,
```
Comment (stevekuznetsov) [inline prow/plugins/skip/skip_test.go]: I think it would be better to have this as a separate case?

Comment (adshmh) [reply prow/plugins/skip/skip_test.go]: Thank you for the review. I have updated the PR.

**Comment by stevekuznetsov** [comment]
Cool! Thanks @adshmh 

/lgtm
/approve

**Pull request merged**